idl/chromadb/proto/chroma.proto

```diff
@@ -87,19 +87,59 @@ message OperationRecord {
     Operation operation = 4;
 }
 
-message VectorEmbeddingRecord {
-    string id = 1;
-    Vector vector = 3; // TODO: we need to rethink source of truth for vector dimensionality and encoding
+/* Metadata Reader Interface */
+
+service MetadataReader {
+    rpc GetMetadata(GetMetadataRequest) returns (GetMetadataResponse) {}
 }
 
-message VectorQueryResult {
-    string id = 1;
-    float distance = 3;
-    optional Vector vector = 4;
+message GetMetadataRequest {
+    Where where = 1;
+    WhereDocument where_document = 2;
+    repeated string ids = 3;
+    optional int32 limit = 4;
+    optional int32 offset = 5;
 }
 
-message VectorQueryResults {
-    repeated VectorQueryResult results = 1;
+enum WhereListType {
+    IN = 0;
+    NIN = 1;
+}
+
+enum WhereChildrenType {
+    AND = 0;
+    OR = 1;
+}
+
+// A `Where` clause for filtering metadata. A `Where` clause is a tree of
+// `Where` clauses, where each node is exactly one of:
+// - A leaf node with a key, a list of values, and a list type (IN or NIN)
```

Comment:
are we essentially representing EQ as IN of list = 1? My only concern is the divergence between how we represent this in protos and the actual QL

Reply:
What about GT/LT/NEQ etc?

Reply:
Added